File: org-go/pkg/models/section.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Section struct {
	ID         uuid.UUID `json:"id"                 db:"id"`
	NotebookID uuid.UUID `json:"notebook_id"        db:"notebook_id"`
	Name       string    `json:"name"               db:"name"`
	Position   int       `json:"position,omitempty" db:"position"`
	CreatedAt  time.Time `json:"created_at"         db:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"         db:"updated_at"`
}

File: sigil-frontend/src/api/model/section.ts
// eslint-disable-next-line no-restricted-imports
import dayjs, { Dayjs } from "dayjs"

export interface Section {
  id: string
  notebook_id: string
  name: string
  position: number
  created_at: Dayjs
  updated_at: Dayjs
}

export function fromJson(section: Section): Section {
  return {
    ...section,
    created_at: dayjs(section.created_at),
    updated_at: dayjs(section.updated_at),
  }
}
